### PR TITLE
add codereviewmaster bot

### PR DESCRIPTION
**Pull request by robbiecloset**
Listens for the string `codereviewmaster`.

Understands the following commands:

`codereviewmaster add jcorrigan` || `codereviewmaster add slack:john
gerrit:jcorrigan`

Adds a user to a list of available reviewers.

`codereviewmaster remove jcorrigan`

Removes user from list of available reviewers

`codereviewmaster list`

Shows list of available reviewers

`codereviewmaster g/123456`

Selects a reviewer for the provided gerrit change; adds the reviewer in
gerrit (if gerrit creds are present).

### CODE REVIEW

File: lib/code_review_master/data.rb [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+          'reviewers' => []
+        })
+      end
+      @_config
+    end
+
+    def db
+      unless @_db
+        @_db = DynamoDB.new(botname: 'angelbot')
+        @_db.create_table(table_name, if_not_exist: true)
```
Comment (robbiecloset) [inline lib/code_review_master/data.rb]: I know that this is commented out in the other class that creates a table (or did at one point). Is that what I should do here as well?

File: bots/code_review_master.rb [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    end
+  end
+
+  private
+
+  ######################
+  # Determine requests #
+  ######################
+  def add_reviewer_request?(message)
+    (message.downcase.match(/^#{CODEREVIEWMASTER} #{ADD_COMMAND} (.+)/) || [])[1]
```
Comment (opes) [inline bots/code_review_master.rb]: You can just use the `i` option in the regex match for case insensitivity so you don't have to `downcase` the message every time.

File: lib/code_review_master/data.rb (hunk 1 above)
Comment (opes) [reply lib/code_review_master/data.rb]: Yeah, we'll need to create the table manually from the looks of it. The `botname` argument in the initialization should probably be `codereviewmaster` though.

Comment (robbiecloset) [reply lib/code_review_master/data.rb]: I am fine with updating the bot name. The other db integration just used the general `angelbot` so I was just following that. Make sense to differentiate though.

File: bots/code_review_master.rb (hunk 2 above)
Comment (robbiecloset) [reply bots/code_review_master.rb]: 👍

File: bots/code_review_master.rb [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            m.send_message("Can't add a user with string #{message_segment} :fail:")
+          end
+        end
+
+        ###################
+        # Remove reviewer #
+        ###################
+        if (to_remove = remove_reviewer_request?(message))
+          SlackbotFrd::Log.debug(
+            "user '#{user}' in channel '#{channel}' is adding the reviewer #{to_remove}"
```
Comment (opes) [inline bots/code_review_master.rb]: s/adding/removing?

File: bots/code_review_master.rb [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        if help_output_request?(message)
+          SlackbotFrd::Log.debug(
+            "user '#{user}' in channel '#{channel}' is requesting help output"
+          )
+          help = <<-HELP_TEXT
+Hi! I help assign code reviewers.
+
+*Assign a reviewer*
+
+`codereviewmaster g/123456`
```
Comment (opes) [inline bots/code_review_master.rb:93]: Nitpick and not required for merge: It's a little confusing as to who's being added as a reviewer. Might want to talk about how it adds a group of users that you specifically define as documented below.

File: lib/code_review_master/data.rb [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        @_config = (item || {
+          'reviewers' => []
+        })
+      end
+      @_config
+    end
+
+    def db
+      unless @_db
+        @_db = DynamoDB.new(botname: 'angelbot')
```
Comment (opes) [inline lib/code_review_master/data.rb]: s/angelbot/codereviewmaster?

**Review by opes — COMMENTED** [review]
Mostly nitpicks, but one to the Dynamo botname would be worthwhile.

**Review by opes — APPROVED** [review]
Looks good! I'll get the Dynamo table created for this and get it deployed as soon as I can.

**Pull request merged**